Experiment consists of 3 doors, behind random one tester puts the money and player selects one of the doors. In this paradox it is claimed if player selection is changed when one of the not selected doors (but without money!) is opened, the probability to win (i.e. pick the door with money) is 2/3. Let's check it down below.

In [23]:
import random 

n = 10000
wins_when_changed = 0

for _ in range(n):
    # tester puts money behind one of the doors
    door_with_money = random.randint(0, 2)
    
    # player selects random door
    player_selection = random.randint(0, 2)
    
    # tester opens the door not selected by player and not containing money
    # if player selected door with money, door chosen randomly
    doors_to_open = [x for x in range(3) if x != player_selection and x != door_with_money]
    if len(doors_to_open) == 1:
        tester_selection = doors_to_open[0]
    else:
        tester_selection = doors_to_open[random.randint(0, 1)]
        
    # change the player's initial selection
    player_new_selection = [x for x in range(3) if x != tester_selection and x != player_selection][0]
    
    # increase wins if matched
    if player_new_selection == door_with_money:
        wins_when_changed += 1

In [24]:
wins_when_changed / n

0.6574

Results are distributed around 0.66 == 2/3, therefore, changing doors indeed increases your chanses to success